### DISCUSSION TITLE
change duckduckgo to startx ?

### DISCUSSION THREAD

**Discussion by r3k2** [qutebrowser/qutebrowser#5684] [Q&A] [answered]
hello I been using duckduckgo for years ! but this last 3 months is been awfull! like very awfull! I am super fustrated with it to the point sometimes I have to open google grgrgr and same search gives me my result... I found that searchx is working for me as duckduckgo used to, actually better. I been reading the help and cant find how to change this....

**Comment by mschilli87** [comment, marked as answer]
Assuming you are a) not using a [`config.py`](https://qutebrowser.org/doc/help/configuring.html#configpy) (if you don't know you probably don't) or b)  it contains `config.load_autoconfig()`, this should do:

1. Open qutebrowser.
2. Press `Esc` to make sure you are in normal mode.
3. Enter command mode by typing a colon (`:`).
4. Run `set url.searchengines {"DEFAULT": "https://searx.xyz/?q={}"}`

Obviuously you can change the exact URL to whatever [searx instance](https://searx.space) you prefer.

---
Note that I usally figure things like this out by myself quickly by looking through [qute://settings/](qute://settings/):

If you open that address in qutebrowser and search - via forward slash (`/`) - for `duckduckggo`, you'd quickly find the following:

> **`url.searchengines` (Current: `{"DEFAULT": "https://duckduckgo.com/?q={}"}`)**
> Search engines which can be used via the address bar.
> 
> Maps a search engine name (such as `DEFAULT`, or `ddg`) to a URL with a
> `{}` placeholder. The placeholder will be replaced by the search term, use
> `{{` and `}}` for literal `{`/`}` braces.
> 
> The following further placeholds are defined to configure how special
> characters in the search terms are replaced by safe characters (called
> 'quoting'):
> 
> * `{}` and `{semiquoted}` quote everything except slashes; this is the most
> sensible choice for almost all search engines (for the search term
> `slash/and&amp` this placeholder expands to `slash/and%26amp`).
> * `{quoted}` quotes all characters (for `slash/and&amp` this placeholder
> expands to `slash%2Fand%26amp`).
> * `{unquoted}` quotes nothing (for `slash/and&amp` this placeholder
> expands to `slash/and&amp`).
> 
> The search engine named `DEFAULT` is used when `url.auto_search` is turned
> on and something else than a URL was entered to be opened. Other search
> engines can be used by prepending the search engine name to the search
> term, e.g. `:open google qutebrowser`.

**Reply by The-Compiler** [reply]
Almost - you'll need to quote the value because it has a space in it:

```
:set url.searchengines '{"DEFAULT": "https://searx.xyz/?q={}"}'
```

However, you can also use `:config-dict-add` instead to make this a bit more convenient:

```
:config-dict-add --replace url.searchengines DEFAULT https://searx.xyz/?q={}
```

**Comment by r3k2** [comment]
hahaha I wrote this before going to bed, just notice I wrote startx :D :D :D good thing you guys figure out I meant searchX :D 
Thanks everyone this seems to be what I was looking for, I searched the manual and site and did not see any info and it was late.
